src/Transpiler.ts

```diff
@@ -1596,13 +1601,16 @@ export abstract class LuaTranspiler {
     }
 
     // Transpile a class declaration
-    public transpileClass(node: ts.ClassDeclaration): string {
-        if (!node.name) {
+    public transpileClass(node: ts.ClassLikeDeclarationBase, nameOverride?: string): string {
+        let className: string;
```

Comment:
`const className = node.name ? this.transpileIdentifier(node.name) : nameOverride;` is much nicer in my opinion.